Adamant/Modules/Chat/View/Subviews/BadgeView.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  BadgeView.swift
+//  Adamant
+//
+//  Created by Аркадий Торвальдс on 26.08.2024.
+//  Copyright © 2024 Adamant. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import SnapKit
+
+class BadgeView: UIView {
+    private lazy var counterLabel: UILabel = {
+        let rv = UILabel()
+        rv.numberOfLines = 0
+        rv.lineBreakMode = .byWordWrapping
+        rv.font = .systemFont(ofSize: 12)
+        rv.textColor = labelColor
+        rv.text = String.empty
+        rv.translatesAutoresizingMaskIntoConstraints = false
+        return rv
+    }()
+    
+    private var labelColor: UIColor {
+        UIColor { traits -> UIColor in
+            return traits.userInterfaceStyle == .dark ? UIColor.adamant.first : UIColor.adamant.fourth
+        }
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        configure()
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        configure()
+    }
+    
+    @MainActor
+    func updateCounter(count: Int) {
+        self.isHidden = count == 0
+        let formatText = formatNumber(count)
+        counterLabel.text = formatText
+    }
+}
+
+private extension BadgeView {
+    func configure() {
+        backgroundColor = UIColor.adamant.primary
+        layer.cornerRadius = 8
+        addSubview(counterLabel)
+        let padding: CGFloat = 4
```

Comment:
Minor. We usually move all the constants to the end of the file as private variables (outside of classes)